adagrams/game.py

```diff
@@ -1,11 +1,186 @@
+import random
+
+letter_quant_dict = {
+    "A" : 9,
+    "B" : 2,
+    "C" : 2,
+    "D" : 4,
+    "E" : 12,
+    "F" : 2,
+    "G" : 3,
+    "H" : 2,
+    "I" : 9,
+    "J" : 1,
+    "K" : 1,
+    "L" : 4,
+    "M" : 2,
+    "N" : 6,
+    "O" : 8,
+    "P" : 2,
+    "Q" : 1,
+    "R" : 6,
+    "S" : 4,
+    "T" : 6,
+    "U" : 4,
+    "V" : 2,
+    "W" : 2,
+    "X" : 1,
+    "Y" : 1,
+    "Z" : 1
+}
+
+letter_score_dict = {
+    "A" : 1,
+    "B" : 3,
+    "C" : 3,
+    "D" : 2,
+    "E" : 1,
+    "F" : 4,
+    "G" : 2,
+    "H" : 4,
+    "I" : 1,
+    "J" : 8,
+    "K" : 5,
+    "L" : 1,
+    "M" : 3,
+    "N" : 1,
+    "O" : 1,
+    "P" : 3,
+    "Q" : 10,
+    "R" : 1,
+    "S" : 1,
+    "T" : 1,
+    "U" : 1,
+    "V" : 4,
+    "W" : 4,
+    "X" : 8,
+    "Y" : 4,
+    "Z" : 10
+}
+
+# # Create a pool from the dictionary
+# def create_pool(letter_quant_dict):
+#     pool = []
+#     for letter in letter_quant_dict:
+#         quant = letter_quant_dict[letter]
+#         while quant > 0:
+#             pool.append(letter)
+#             quant -= 1
+#     return pool
+# pool = create_pool(letter_quant_dict)
+
 def draw_letters():
-    pass
+    pool = []
+    for letter in letter_quant_dict:
+        quant = letter_quant_dict[letter]
+        while quant > 0:
+            pool.append(letter)
+            quant -= 1
+    hand = []
+    draw = 10
+    while draw > 0:
+        index = random.randint(0, len(pool)-1)
+        hand.append(pool[index])
+        pool.pop(index)
+        draw -= 1
+    return hand
+
+# def uses_available_letters(word, letter_bank):
+#     hand = letter_bank
+#     for letter in word.upper():
+#         if not letter in hand:
+#             return False
+#         hand.remove(letter)
+#     return True
 
 def uses_available_letters(word, letter_bank):
-    pass
+    #I dont understand why we woulndt want to update the letter_bank but ok
+
+    #build letter_bank_dict
+    letter_bank_dict = {}
+    for letter in letter_bank:
+        if letter in letter_bank_dict:
+            letter_bank_dict[letter] += 1
+        else:
+            letter_bank_dict[letter] = 1
+
+    #build hand_dict
+    hand_dict = {}
+    for letter in word.upper():
+        if letter in hand_dict:
+            hand_dict[letter] += 1
+        else:
+            hand_dict[letter] = 1
+    
+    #compare two dictionaries
+    for letter in hand_dict:
+        if letter not in letter_bank_dict:
+            return False
+        elif hand_dict[letter] > letter_bank_dict[letter]:
+            return False
+    return True
+
 
 def score_word(word):
-    pass
+    word_score = 0
+    if len(word) >= 7:
+        word_score += 8
+    for letter in word.upper():
+        letter_score = letter_score_dict[letter]
+        word_score += letter_score
```

Comment:
✅ Great approach to look up the score for each letter and accumulate it into the result.